overlord/configstate/configcore/journal.go

```diff
@@ -0,0 +1,108 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/osutil/sys"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/systemd"
+)
+
+var osutilFindGid = osutil.FindGid
+var sysChownPath = sys.ChownPath
+
+func init() {
+	supportedConfigurations["core.journal.persistent"] = true
+}
+
+func validateJournalSettings(tr config.ConfGetter) error {
+	return validateBoolFlag(tr, "journal.persistent")
+}
+
+func handleJournalConfiguration(tr config.ConfGetter, opts *fsOnlyContext) error {
+	output, err := coreCfg(tr, "journal.persistent")
+	if err != nil {
+		return nil
+	}
+
+	if output == "" {
+		return nil
+	}
+
+	var sysd systemd.Systemd
+	rootDir := dirs.GlobalRootDir
+	if opts != nil {
+		rootDir = opts.RootDir
+	}
+
+	tempLogPath := filepath.Join(rootDir, "/var/log/journal-snapd")
```

Comment:
I wonder if for tidyness we should remove this always when entering this function, but maybe it's not worth it